browser/ntp_sponsored_images/new_tab_page_branded_view_counter.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_NTP_SPONSORED_IMAGES_NEW_TAB_PAGE_BRANDED_VIEW_COUNTER_H_
+#define BRAVE_BROWSER_NTP_SPONSORED_IMAGES_NEW_TAB_PAGE_BRANDED_VIEW_COUNTER_H_
+
+#include <memory>
+
+#include "components/keyed_service/core/keyed_service.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "brave/browser/ntp_sponsored_images/view_counter_model.h"
+#include "brave/components/brave_rewards/browser/rewards_service_observer.h"
+#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_images_component_manager.h"
+#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_images_data.h"
+
+class Profile;
+
+class NewTabPageBrandedViewCounter
+    : public KeyedService,
+      public NTPSponsoredImagesComponentManager::Observer,
+      public brave_rewards::RewardsServiceObserver {
+ public:
+  static NewTabPageBrandedViewCounter* GetForProfile(Profile* profile);
+
+  explicit NewTabPageBrandedViewCounter(Profile* profile);
+  ~NewTabPageBrandedViewCounter() override;
+
+  NewTabPageBrandedViewCounter(const NewTabPageBrandedViewCounter&) = delete;
+  NewTabPageBrandedViewCounter& operator=(
+      const NewTabPageBrandedViewCounter&) = delete;
+
+  // KeyedService
+  void Shutdown() override;
+
+  // NTPSponsoredImagesComponentManager::Observer
+  void OnUpdated(const NTPSponsoredImagesData& data) override;
```

Comment:
these observer methods probably shouldn't be public on this class

Reply:
fixed